src/aegis/src/aegis_example_key_manager.erl

```diff
@@ -0,0 +1,55 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(aegis_example_key_manager).
+
+
+-behaviour(aegis_key_manager).
+
+
+-export([
+    init/0,
+    generate_key/3,
+    unwrap_key/3
+]).
+
+
+
+init() ->
+    case config:get("aegis_example_key_manager", "key_provider") of
+        undefined ->
+            erlang:error(misconfigured_key_provider);
+        KeyProvider ->
+            PassPhrase = os:cmd(KeyProvider),
+            {ok, RootKey} = couch_passwords:pbkdf2(
+                iolist_to_binary(PassPhrase), <<0:256>>, 10000, 32),
+            <<(binary_to_integer(RootKey, 16)):256>>
```

Comment:
even as an example not applying salt is a problem, I'm also confused by the construction of line 35, is it a reluctance to modify couch_passwords.erl? We really want sha256 involved in the derivation not sha1. Should I take on that piece?

Reply:
This is because `couch_passwords:pbkdf2/4` returns value in hex format, so this line converts it back.

I agree that modifying this in `couch_passwords.erl` is much cleaner. Unfortunately I'm not familiar with that part of codebase, so can I take you up on the offer, please?

Reply:
agreed on call to delete this module and come back to it after GA